### DISCUSSION TITLE
Dynamic grid values on Repeater Items

### DISCUSSION THREAD

**Discussion by KONEY** [filamentphp/filament#13791] [Help]
### Package

Form builder

### Package Version

v3.0.0

### How can we help you?

I'm building a form with questions from a DB, organized with a Repeater with a grid of 1. Some items need to be coupled on a grid of 2 when they have true on a specific field. However, I can't pass anything to repeater grid like this:`->grid(function (){		return 2;		})` because it triggers an error: 

> Filament\Forms\Components\Repeater::grid(): Argument #1 ($columns) must be of type array|string|int|null, Closure given

Also the repeater does $record does not contain the value of the item, so I'm not sure how to pass it to the grid, if possible at all...

Any help would be appreciated, also suggestions for alternative solutions!

thanks

**Comment by leandrocfe** [comment]
hum, I think you can't do this

**Reply by KONEY** [reply]
As I suspected... but any other options?

**Comment by KONEY** [comment]
I solved using extraAttributes() to add different css classes for single or coupled items and adding customizations to the stylesheet.
